src/arrayMethodReduce.js

```diff
@@ -5,7 +5,22 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    const areBothArgumentPassed = arguments.length === 2;
+    let total = this[0];
+
+    if (typeof initialValue === 'number') {
+      total = initialValue;
+    } else if (areBothArgumentPassed) {
+      total = initialValue + '';
+    }
+
+    const startValue = areBothArgumentPassed ? 0 : 1;
```

Comment:
More descriptive variable name
```suggestion
    const startIndex= areBothArgumentPassed ? 0 : 1;
```